src/models/StyledComponent.js

```diff
@@ -38,18 +37,38 @@ export default (ComponentStyle: Function, constructWithOptions: Function) => {
       : componentId
   }
 
-  class BaseStyledComponent extends AbstractStyledComponent {
+  class BaseStyledComponent extends Component {
     static target: Target
     static styledComponentId: string
     static attrs: Object
     static componentStyle: Object
     static warnTooManyClasses: Function
 
+    static contextTypes = {
+      [CHANNEL]: PropTypes.func,
+      [CHANNEL_NEXT]: CONTEXT_CHANNEL_SHAPE,
+      [CONTEXT_KEY]: PropTypes.instanceOf(StyleSheet),
+    }
+
+    static propTypes = {
```

Comment:
Sorry, I just caught some more things. Can you remove the `contextTypes` and `propTypes` static property from the `BaseStyledComponent` please? :wink: they'll only need to be present on each `StyledComponent`.

Also, let's not add the `propTypes`. It has worked out fine without them before fortunately, and it might break some things and output unwanted warnings for some people. Even if it's the right thing to do, it's something we should save up for the major bump, as people really become annoyed when new warnings surface (even if they're warranted)